sqlalchemy_filterset/strategies.py

```diff
@@ -52,7 +62,21 @@ def _join_if_necessary(self, query: Select) -> Select:
         return query
 
     def _build_join(self, query: Select, onclause: ColumnElement[bool]) -> Select:
-        return query.join(self.model, onclause=onclause)
+        return query.join(self.model, onclause=onclause, isouter=self.is_outer, full=self.is_full)
+
+
+class JoinChainStrategy(BaseStrategy):
+    def __init__(
+        self,
+        *chain: RelationJoinStrategy,
+    ) -> None:
+        self.chain = chain
+
+    def filter(self, query: Select, expression: Any) -> Select:
+        query = functools.reduce(
+            lambda query, strategy: strategy.apply_join(query), self.chain, query
+        )
+        return query.where(expression)
 
```

Comment:
Maybe rename 
JoinChainStrategy to MultiJoinStrategy
RelationJoinStrategy to JoinStrategy
RelationSubqueryExistsStrategy to ExistsStrategy

as @RamiAwar suggested in https://github.com/sqlalchemy-filterset/sqlalchemy-filterset/issues/11#issuecomment-2219629117